lib/Dialects/LinalgExt/Transforms/TilingToTileOp.cpp

```diff
@@ -0,0 +1,131 @@
+  assert(loopRanges.size() >= 1 &&
+         "expected at least a single loop in operation");
+  auto tileOp = rewriter.create<linalg_ext::TileOp>(
+      loc, tileSize, op.getDestinationOperands(rewriter),
+      [&](OpBuilder &b, Location loc, Value offset, Value size,
+          ValueRange outSlices) {
+        // TODO: support `getTiledImplementation` with >1 produced tiled ops.
+        SmallVector<OpFoldResult> tiledOffsets = {offset}, tiledSizes = {size};
+        for (unsigned i = 1; i < loopRanges.size(); ++i) {
+          tiledOffsets.push_back(loopRanges[i].offset);
+          tiledSizes.push_back(loopRanges[i].size);
+        }
+        Operation *tiledOp =
+            op.getTiledImplementation(b, outSlices, tiledOffsets, tiledSizes);
+        b.create<linalg_ext::TileYieldOp>(loc, tiledOp->getResults());
+      });
+  return tileOp->getResults();
+}
+
+namespace {
+
+struct OpTilingPattern : public OpInterfaceRewritePattern<TilingInterface> {
+  OpTilingPattern(MLIRContext *context, linalg::LinalgTilingOptions opt)
+      : OpInterfaceRewritePattern<TilingInterface>(context), options(opt) {}
+
+  LogicalResult matchAndRewrite(TilingInterface op,
+                                PatternRewriter &rewriter) const override {
+    // Poor man's single pattern application to avoid infinite pattern
+    // application wihtout having to hack with attributes and filters.
+    if (successfulApplicationCount <= 0)
+      return failure();
+
+    /// Currently only handle single result operations.
+    if (op->getNumResults() != 1)
+      return failure();
+
+    // Get rank and tile sizes.
+    SmallVector<Value> tileSizes =
+        options.tileSizeComputationFunction(rewriter, op);
+    assert(tileSizes.size() == 1 && "expected a single tile size");
+
+    /// Currently only handle operations with all parallel iterator types.
+    auto loopIteratorTypes = op.getLoopIteratorTypes();
+    if (loopIteratorTypes.empty()) {
+      // Scalar operation, nothing to do, so just return.
+      return failure();
+    }
+    ArrayRef<StringRef> loopIteratorTypesRef(loopIteratorTypes);
+    if (llvm::any_of(loopIteratorTypesRef.take_front(tileSizes.size()),
+                     [](StringRef iteratorType) {
+                       return iteratorType != getParallelIteratorTypeName();
+                     })) {
+      return failure();
+    }
+
+    rewriter.replaceOp(op, tileToTileOp(rewriter, op, tileSizes.front()));
+
+    // Poor man's single pattern application to avoid infinite pattern
+    // application wihtout having to hack with attributes and filters.
+    --successfulApplicationCount;
+
+    return success();
+  }
+
+private:
+  linalg::LinalgTilingOptions options;
+  linalg::LinalgTransformationFilter filter;
+  // Poor man's single pattern application to avoid infinite pattern
+  // application wihtout having to hack with attributes and filters.
+  mutable int64_t successfulApplicationCount = 1;
```

Comment:
Since you're decrementing perhaps rename to `remainingApplicationCount`? `successfulApplicationCount` sounds like it's counting up